dashboard/public/openapi.yaml

```diff
@@ -0,0 +1,1901 @@
+                type: string
+              url:
+                type: string
+        name:
+          type: string
+        sources:
+          type: array
+          items: 
+            type: string
+        tags:
+          type: string
+        version:
+          type: string
+    Release:
+      type: object
+      properties:
+        name:
+          type: string
+        info:
+          type: string
+        chart:
+          type: object
+        config:
+          type: object
+        manifest:
+          type: string
+        hooks:
+          type: array
+          items:
+            type: object
+        version:
+          type: integer
+        namespace:
+          type: string
+    ChartCategory:
+      type: object
+      properties:
+        name:
+          type: string
+        count:
+          type: integer
+    Chart:
+      type: object
+      properties:
+        ID:
+          type: integer
+        name:
+          type: string
+        repo:
+          $ref: '#/components/schemas/Repo'
+        description:
+          type: string
+        home:
+          type: string
+        keywords:
+          type: array
+          items:
+            type: string
+        maintainers:
+          type: array
+          items:
+            type: object
+        sources:
+          type: array
+          items:
+            type: string
+        icon:
+          type: string
+        raw_icon:
+          type: string
+          format: byte
+        icon_content_type:
+          type: string
+        category:
+          type: string
+        chartVersions:
+          type: object
+          items:
+            type: string
+    ChartVersion:
+      type: object
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attibutes:
+          type: object
+          properties:
+            version:
+              type: string
+            app_version:
+              type: string
+            created:
+              type: string
+              format: time
+            digest:
+              type: string
+            urls:
+              type: array
+              items:
+                type: string
+            readme:
+              type: string
+            values:
+              type: string
+            schema:
+              type: string
+        links:
+          type: object
+        relationships:
+          type: object
+          properties:
+            data:
+              type: object
+            links:
+              type: object
+              properties:
+                self:
+                  type: string
+    Repo:
+      type: object
+      properties:
+        namespace:
+          type: string
+        name:
+          type: string
+        url:
+          type: string
+        type:
+          type: string
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+security:
+  - bearerAuth: []
```

Comment:
Woah - what did you use to generate and manually assemble that?

Reply:
Currently, it has been a full-manual process using the types already defined in go :grimacing: 
There a couple of tools for doing some automation (https://github.com/go-swagger/go-swagger or https://github.com/swaggo/swag) but given how we are defining the APIs (multiple places, params fuzzily defined, proxypassing with nginx, etc) probably it won't work... I mean, it's something I want to check, but don't think it'll work.

Reply:
Wow, that's 2k of manual openapi.yaml - impressive :)